encodedcc.py

```diff
@@ -662,8 +662,18 @@ def patch_set(args, connection):
                     if k[1] == "int" or k[1] == "integer":
                         patch_data[k[0]] = int(temp_data[key])
                     elif k[1] == "array" or k[1] == "list":
+                        if type(temp_data[key]) == str:
+                            # So JSON loads if present.
+                            temp_data[key] = temp_data[key].replace("'",'"')
+                        # Try to convert string before testing type. 
+                        try:
+                            temp_data[key] = json.loads(temp_data[key])
+                        except:
+                            pass
```

Comment:
If it is a flat :list or :array it is still treated as a string and there is some weird parsing code in the final else statement that deals with it: `else:
                            l = temp_data[key].strip("[]").split(", ")
                            l = [x.replace("'", "") for x in l]`

Trying to parse a string that doesn't have JSON in it doesn't work. In the case of patch_set it is probably a flat list and should be let through.
<img width="620" alt="screen shot 2017-08-01 at 2 49 22 pm" src="https://user-images.githubusercontent.com/16053922/28848670-b3eaa1d8-76c8-11e7-8a22-8e7fb8114dd4.png">